MIGRATING.md

```diff
@@ -3,111 +3,104 @@ Migrating
-for commands.
+Version 2.0.0 simplifies the `CommandInterface` and adds an `OptionsInterface`.
+Combined, these two interfaces represent the fundamental requirements of a
+command system that is flexible and can be used in modern and legacy environments.
 
-### Updating Commands
+### Creating an Options Value Object
 
-Commands that used to be defined as:
+All options must implement `OptionsInterface`, which includes the ability to
+serialize the value object to JSON. The `OptionsSerializerTrait` can provide
+this functionality.
 
-```php
-use Equip\Command\AbstractCommand;
-
-class LoginCommand extends AbstractCommand
-{
-    // ...
-}
-```
-
-Should now be defined as:
+Options must validate the values passed during construction. If you are using an
+array to hydrate the options, the `OptionsRequiredTrait` will help you validate
+requirements. Type checking should also be done when suitable.
 
 ```php
-use Equip\Command\Command;
-
-class LoginCommand extends Command
-{
-    // ...
-}
-```
-
-This will also require modification in how commands use options internally.
-Previously the `execute()` method would access options as an array:
+use Equip\Command\OptionsInterface;
+use Equip\Command\OptionsRequiredTrait;
+use Equip\Command\OptionsSerializerTrait;
 
-```php
-public function execute()
+class LoginOptions implements OptionsInterface
 {
-    $options = $this->options();
-
-    $this->checkCredentials($options['email'], $options['password']);
+    use OptionsRequiredTrait;
+    use OptionsSerializerTrait;
 
-    // ...
-}
-```
+    private $email;
+    private $password;
 
-Instead, the options will now be a read-only object:
+    public function __construct(array $options)
+    {
+        $this->checkRequired($options, [
+            'email',
+            'password',
+        ]);
 
-```
-public function execute()
-{
-    $options = $this->options();
+        $this->email = $options['email'];
+        $this->password = $options['password'];
+    }
 
-    $this->checkCredentials($options->email, $options->password);
+    public function email()
+    {
+        return $this->email;
+    }
 
-    // ...
+    public function password()
+    {
+        return $this->password;
+    }
 }
 ```
 
-Any attempt to modify the options will throw an `ImmutableException`.
+### Creating a Command
 
-### Creating Options
-
-Command options are now defined as separate classes. These classes must extend
-the `Options` class:
+All commands must implement the `CommandInterface` instead of `AbstractCommand`.
 
 ```php
-use Equip\Command\Options;
+use Equip\Command\CommandInterface;
 
-final class LoginOptions extends Options
+class LoginCommand implements CommandInterface
 {
-    protected $email;
-    protected $password;
-    protected $remember = false;
-
     /**
-     * @inheritdoc
+     * @var LoginOptions
      */
-    public function required()
+    private $options;
+
+    public function withOptions(LoginOptions $options)
```

Comment:
Is this exact logic likely to be the most common use, or would `setOptions` often be customized (for validation purposes or otherwise)? That strikes me as the main question here.